dataeng/resources/detect-new-dbt-manual-models-fields.sh

```diff
@@ -0,0 +1,79 @@
+set -ex
+
+# Setup a virtual environment
+PYTHON38_VENV="py38_venv"
+virtualenv --python=python3.8 --clear "${PYTHON38_VENV}"
+source "${PYTHON38_VENV}/bin/activate"
+
+# Go into analytics-tools and install the dependencies
+cd ${WORKSPACE}/analytics-tools/snowflake
+make requirements
+
+# Fetch credentials from vault
+# Do not print commands in this function since they may contain secrets.
+set +x
+
+# Retrieve a vault token corresponding to the jenkins AppRole.  The token is then stored in the VAULT_TOKEN variable
+# which is implicitly used by subsequent vault commands within this script.
+# Instructions followed: https://learn.hashicorp.com/tutorials/vault/approle#step-4-login-with-roleid-secretid
+export VAULT_TOKEN=$(vault write -field=token auth/approle/login \
+    role_id=${ANALYTICS_VAULT_ROLE_ID} \
+    secret_id=${ANALYTICS_VAULT_SECRET_ID}
+)
+
+# JIRA webhook URL and secret string from vault
+WEBHOOK_URL=$(
+  vault kv get \
+    -version=${JIRA_WEBHOOK_VAULT_KV_VERSION} \
+    -field=JIRA_WEBHOOK_URL \
+    ${JIRA_WEBHOOK_VAULT_KV_PATH} \
+)
+WEBHOOK_SECRET=$(
+  vault kv get \
+    -version=${JIRA_WEBHOOK_VAULT_KV_VERSION} \
+    -field=JIRA_WEBHOOK_SECRET \
+    ${JIRA_WEBHOOK_VAULT_KV_PATH} \
+)
+
+# Snowflake credentials from vault
+SNOWFLAKE_ACCOUNT=$(
+  vault kv get \
+    -version=${AUTOMATION_TASK_VAULT_KV_VERSION} \
+    -field=account \
+    ${AUTOMATION_TASK_VAULT_KV_PATH} \
+)
+
+SNOWFLAKE_USER=$(
+  vault kv get \
+    -version=${AUTOMATION_TASK_VAULT_KV_VERSION} \
+    -field=user \
+    ${AUTOMATION_TASK_VAULT_KV_PATH} \
+)
+# The detect_new_raw_columns.py script, much like all other scripts that connect to Snowflake,
+# expects the private key and the privarte key passphrase to be in files.
+# As a result, SNOWFLAKE_PRIVATE_KEY and SNOWFLAKE_PRIVATE_KEY_PASSPHRASE are stored in files.
+vault kv get \
+    -version=${AUTOMATION_TASK_VAULT_KV_VERSION} \
+    -field=private_key \
+    ${AUTOMATION_TASK_VAULT_KV_PATH} > .private_key_file
+
+vault kv get \
+    -version=${AUTOMATION_TASK_VAULT_KV_VERSION} \
+    -field=private_key_passphrase \
+    ${AUTOMATION_TASK_VAULT_KV_PATH} > .private_key_passphrase_file
+set -x
+
+# The extra vars file for this job contains both field mappings and the necessary credentials for Snowflake and Jenkins.
+# Therefore, the options to the script are read from the config file.
+CONFIG_PATH=${WORKSPACE}/analytics-secure/job-configs/DETECT_NEW_DBT_MANUAL_MODELS_FIELDS_JOB_MAPPINGS.yaml
+
+# Invoke the script to detect new fields that need to be added manually
+python detect_new_raw_columns.py ${CONFIG_PATH} \
+    --user ${SNOWFLAKE_USER} --account ${SNOWFLAKE_ACCOUNT} \
+    --key-path .private_key_file --passphrase-path .private_key_passphrase_file \
+    --jira-webhook-url ${WEBHOOK_URL} \
+    --jira-webhook-secret ${WEBHOOK_SECRET}
+
+# Clean up the temporary files with SNOWFLAKE_PRIVATE_KEY and SNOWFLAKE_PRIVATE_KEY_PASSPHRASE
+rm -rf .private_key_file .private_key_passphrase_file
```

Comment:
Could you place this cleanup logic in a bash function, and add a bash trap on EXIT?  E.g.: https://github.com/edx/edx-analytics-configuration/blob/a446263db34b901d90504fc706a507febdc4a3cf/automation/run-automated-task.sh#L25-L31

This helps to ensure that `rm` gets run, except in the most extreme (unlikely) case of `kill -9`/SIGKILL